### ISSUE TITLE
Elastic tube 3D

### ISSUE THREAD

**Issue by davydden** [precice/tutorials#344]
Hi all,

i tried running elastic tube 3D locally with calculix and OpenFoam.
Everything works. What puzzles me is that results https://precice.org/tutorials-elastic-tube-3d.html are rather different from those reported by Valdez PhD [1] for the same case, see Figure 5.28. The difference is in two order of magnitude by eye-balling measure.
As far as i can see, everything is the same in terms of material properties.
It could be that the geometry is different, but this would be surprising.

This case is used in many publications on QN, but i have not found anything but [1] where values at certain point would be plotted.

- Tutorials state (last commit / release): v202211
- Versions used:
  - openfoam@2206
  - precice@2.5.0 build_type=Debug
  - of-precice@1.2.2
  - calculix@2.20

[1] https://upcommons.upc.edu/bitstream/handle/2117/94177/01JGVV01de01.pdf;sequence=1

**Comment by davydden** [comment]
pinging @KyleDavisSA and @uekerman

**Comment by davidscn** [comment]
Hi Denis,

the main 'difference' your pointing out here is probably the associated coupling data used in the scenarios and watchpoints: the tutorial uses [`DisplacementDeltas`](https://github.com/precice/tutorials/blob/0482e5b4d3e1604dd031d2b113a2aaa63733c9a8/elastic-tube-3d/precice-config.xml#L23), which refer to incremental displacements. I could imagine that the data on the website is mostly meant for comparison to the tutorial setup. Using absolute displacements and a reference point of `0.5 L` (as in the tutorial/the reference above as well), I get the following results 

![image](https://github.com/precice/tutorials/assets/33414590/82699729-3796-47f6-8b87-65c4dc8ebae7)

(tried it here with deal.II, but CalculiX should give the same results). This was a rather coarse solid resolution and by eye ball measure the solution quality should be ok. You can just rerun the case and remove the 'Delta' suffix from the variable names (in the precice-config, the precice OpenFOAM dict and the calculix config) to reproduce the results.

Hope this helps!

**Comment by davydden** [comment]
thank you very much @davidscn . I did not notice that it uses incremental displacement. I will give it a try locally.

**Comment by davydden** [comment]
Hi @davidscn , i was able to re-run this and see a similar plot to what you attached above, which is good!

However when looking at the results in Valdes, i think there is still a rathe large difference, namely the maximum radial displacement ~0.01cm at the middle of the tube is attained around 0.007 seconds (eye-ball measure), whereas here we get there at about 0.009seconds. I wonder whether one possibly need to tell OpenFoam somewhere that the passed displacement are total (w.r.t undeformed configuration) as opposed to incremental?
As an additional point of comparison it would be interesting to plot pressure on fluid side, but i don't know Openfoam in order to add that output. Perhaps it's easier done via FSI traction on Calculix, what do you think?

**Comment by davidscn** [comment]
> . I wonder whether one possibly need to tell OpenFoam somewhere that the passed displacement are total (w.r.t undeformed configuration) as opposed to incremental?

No, OpenFOAM deduces the data meaning from the data name, so renaming it should be sufficient.

> As an additional point of comparison it would be interesting to plot pressure on fluid side, but i don't know Openfoam in order to add that output.

In principle, one can extract force information from the watchpoint as well (which is directly related to the pressure).

> , i think there is still a rathe large difference, namely the maximum radial displacement ~0.01cm at the middle of the tube is attained around 0.007 seconds (eye-ball measure), whereas here we get there at about 0.009seconds.

Apart from quite some numerical differences, I could imagine that inlet condition might be different. Looking at the displacement pulse at the 'earlier' location of `0.25 L`, it can be seen that the pulse piles up a bit later in time. Right now, we apply a Dirichlet condition for the pressure (time dependent) and a zeroGradient condition for the velocity. Is this also what they are referring to in the paper?
Do you have any other fluid solver to compare with apart from OpenFOAM?

**Comment by davydden** [comment]
> Right now, we apply a Dirichlet condition for the pressure (time dependent)

that's the same as elsewhere in the literature, so i think what currently done in OpenFOAM https://github.com/precice/tutorials/blob/master/elastic-tube-3d/fluid-openfoam/0/p#L20-L23 
is perfectly fine.

>  zeroGradient condition for the velocity

i don't think it is always possible to impose zeroGradient in additional to zero pressure outlet.

Outlet BC are NOT mentioned in 
Küttler, U. & Wall, W. A. Fixed-point fluid--structure interaction solvers with dynamic relaxation. Computational mechanics, Springer, 2008, 43, 61-72, Section 5.2 
nor in
Degroote, J.; Bathe, K.-J. & Vierendeels, J. Performance of a new partitioned procedure versus a monolithic procedure in fluid-structure interaction. Computers & Structures, Elsevier, 2009, 87, 793-801, Section 4.1

I wonder if this tutorial has the same setup as used in the following preCICE paper

Blom, D.; Lindner, F.; Mehl, M.; Scheufele, K.; Uekermann, B. & van Zuijlen, A. A Review on Fast Quasi-Newton and Accelerated Fixed-Point Iterations for Partitioned Fluid--Structure Interaction Simulation. Advances in Computational Fluid-Structure Interaction and Flow Simulation, Springer, 2016, 257-269

> Do you have any other fluid solver to compare with apart from OpenFOAM?

I can get results which are closer to those presented in Valdez' PhD using a commercial software without preCICE. That's why i was trying to figure out what could the differences, from the first look everything should be the same and everyone cite the "same" elastic tube setup ;-)
